src/MiniCover/Instrumentation/CustomAssemblyResolver.cs

```diff
@@ -1,19 +1,40 @@
-﻿using MiniCover.Utils;
+﻿using Microsoft.Extensions.DependencyModel;
+using MiniCover.Utils;
 using Mono.Cecil;
 using System;
 using System.Collections.Generic;
+using System.IO;
+using System.Linq;
 
 namespace MiniCover.Instrumentation
 {
     public class CustomAssemblyResolver : DefaultAssemblyResolver
     {
-        public CustomAssemblyResolver() : base()
+        private DependencyContext _dependencyContext;
+
+        public CustomAssemblyResolver(string assemblyDirectory)
         {
             RemoveSearchDirectory(".");
             RemoveSearchDirectory("bin");
+
+            AddSearchDirectory(assemblyDirectory);
+
+            _dependencyContext = DepsJsonUtils.LoadDependencyContext(assemblyDirectory);
```

Comment:
I'm adding a DependencyContext, it reads the file `deps.json`. That file contains all the referenced libraries and their paths, with specific version.

Reply:
Nice idea! I like this approach.

Could you point, please, where is assembly version validated?

Reply:
I pointed in another comment.